src/metabase/api/macros.clj

```diff
@@ -129,15 +129,41 @@
             (s/spec (s/cat :path    string?
                            :regexes (s/+ ::defendpoint.route.key-regex-pair))))))
 
+(mu/defn- auto-route-param-regex :- [:maybe (ms/InstanceOfClass java.util.regex.Pattern)]
+  [route-param :- :keyword
+   route-params-schema]
+  (when (= (mc/type route-params-schema) :map)
+    (some (fn [[k _options v-schema]]
+            (when (= k route-param)
+              (or
+               (second (metabase.api.common.internal/->matching-regex v-schema))
+               (:api/regex (mc/properties v-schema)))))
```

Comment:
It seems like :api/regex is more specific, and if you set it, then it should override the one in ->matching-regex. Or at least if you set it and it never gets used we should show a warning or error in dev.

Reply:
yeah you're right good call I'll change it.